cmd/di.go

```diff
@@ -809,3 +821,24 @@ func (di *Dependencies) bootstrapFirewall(options node.OptionsFirewall) error {
 	}
 	return nil
 }
+
+func (di *Dependencies) handleHTTPClientConnections() error {
+	if di.HTTPClient == nil {
+		return errors.New("HTTPClient is not initialized")
+	}
+
+	latestState := connection.NotConnected
```

Comment:
Is it thread-safe to have a kind of global variable here with the state. Can we receive several events that will try to change single variable at the same time?

Reply:
depends if you care about races, as this will race. Subscribe async means that the handler will be called in a goroutine, so if many events get flooded this will have a racy behaviour. 

1) add a lock in the subscribe async
2) consider subscribing in a non async fashion

Reply:
Since I care only about Connected or NotConnected events it would be strange If I get race condition here, but yes, in theory it could happen. Changed to Subscribe().